src/nvim/lua/treesitter.c

```diff
@@ -272,12 +272,18 @@ int tslua_inspect_lang(lua_State *L)
       // not used by the API
       continue;
     }
-    lua_createtable(L, 2, 0);  // [retval, symbols, elem]
-    lua_pushstring(L, ts_language_symbol_name(lang, (TSSymbol)i));
-    lua_rawseti(L, -2, 1);
-    lua_pushboolean(L, t == TSSymbolTypeRegular);
-    lua_rawseti(L, -2, 2);  // [retval, symbols, elem]
-    lua_rawseti(L, -2, (int)i);  // [retval, symbols]
+    const char *name = ts_language_symbol_name(lang, (TSSymbol)i);
+    bool named = t == TSSymbolTypeRegular;
+    lua_pushboolean(L, named);  // [retval, symbols, is_named]
+    if (!named) {
+      size_t new_name_len = strlen(name) + 2;
+      char *new_name = xmallocz(new_name_len);
```

Comment:
I would use stack memory here. Other functions in this file use `char ebuf[BUFSIZE]` where `BUFSIZE=256`.

Also use `vim_snprintf` instead of `snprintf`.

Reply:
> other functions use `vim_snprintf`. Not sure if that is preferred over `snprintf`

I'm not aware of any reason to prefer `vim_snprintf` for cases where `snprintf` works just as well (and that seems to avoid an extra layer). We currently don't have clear guidance on this, but if someone knows better then we can write that down.